Add list of considered but excluded options

thomasnyman opened the pull request:
Closes #146 and #148

siddhesh commented on docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md:
```diff
@@ -783,4 +782,17 @@ Note that the build ID does not act as a checksum for the executable or debug in
+
+| Compiler Flag | Supported since  | Rationale |
+|---------------|------------------|-----------|
+| <span id="-Wl,-z,nodump">`-Wl,-z,nodump`</span>         | - | Single-purpose feature only supported by Solaris linkers[^nodump].
```
"Single-purpose feature for Solaris compatibility" because technically, the binutils linker does implement it but it's useless on Linux.

thomasnyman commented:
Addressed feedback from [C/C++ Compiler BP Guide call 2023-06-21](https://docs.google.com/document/d/1UClGUnOSkOH_wab6Lx43KUdkaK37L8sbWJ_GPZvc1YY/edit#bookmark=kix.nsus8565fs6e) and improved the explanation the footnotes.

thomasnyman commented:
Appease the great linter in the cloud 🙏
